Undeclared dependency to google-http-client-gson

The upgrade from `1.0.11` to `1.3.4` is not working for us. We're running into the following error:

```
Caused by: java.lang.NoClassDefFoundError: com/google/api/client/json/gson/GsonFactory

    at org.postgresql.util.ObjectFactory.instantiate (ObjectFactory.java:43)
    at org.postgresql.core.SocketFactoryFactory.getSocketFactory (SocketFactoryFactory.java:37)
    at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl (ConnectionFactoryImpl.java:122)
    at org.postgresql.core.ConnectionFactory.openConnection (ConnectionFactory.java:49)
    at org.postgresql.jdbc.PgConnection.<init> (PgConnection.java:195)
    at org.postgresql.Driver.makeConnection (Driver.java:454)
    at org.postgresql.Driver.connect (Driver.java:256)
```

This seems to be due to an undeclared dependency to `google-http-client-gson`, which was introduced in [this commit](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/commit/2de5b9e3e3c4b31482462be28926d787c4a6cff8).

I'm not sure if there's anything else that depends on Jackson, but the dependency should probably be changed. I'll try to write a PR when I have a chance.

--- a/core/pom.xml
+++ b/core/pom.xml
@@ -15,9 +15,9 @@
  limitations under the License.
 -->
 
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
@@ -48,6 +48,11 @@
     <dependency>
       <groupId>com.google.http-client</groupId>
       <artifactId>google-http-client-jackson2</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.http-client</groupId>
+      <artifactId>google-http-client-gson</artifactId>
     </dependency>
 
     <dependency>

